src/sinks/aws_s3/config.rs

```diff
@@ -141,13 +169,18 @@ impl S3SinkConfig {
             .filename_append_uuid
             .unwrap_or(DEFAULT_FILENAME_APPEND_UUID);
 
+        let transformer = self.encoding.transformer();
+        let (framer, serializer) = self.encoding.clone().encoding();
+        let framer = framer.unwrap_or_else(|| BytesEncoder::new().into());
```

Comment:
This might not be directly related to this PR, but the naming here is a little confusing. Is `BytesEncoder` a framer? It seems like it, since we create the actual `Encoder` on the next line, so maybe it should be called `BytesFramer`?

Reply:
Agree, that makes more sense. I think that decision came from the fact that "`Framer`" doesn't indicate any direction (encoding/decoding) and at the beginning of all this we weren't sure yet if one struct should serve both directions or not. Opened #12161 with some additional context